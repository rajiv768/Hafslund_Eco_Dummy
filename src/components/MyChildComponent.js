import React from 'react';
import PropTypes from 'prop-types';
import { Segment } from 'semantic-ui-react';

const MyChildComponent = ({ msg, open }) => {
  return (
    <Segment>
      <h3>{msg}</h3>
      {open && <h5>Widget Open</h5>}
    </Segment>
  );
};

MyChildComponent.propTypes = {
  msg: PropTypes.string,
  open: PropTypes.bool,
};

MyChildComponent.defaultProps = {
  msg: 'Hello World',
  open: true,
};

export default MyChildComponent;
