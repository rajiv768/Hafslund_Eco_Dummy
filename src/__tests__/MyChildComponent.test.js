import React from 'react';
import '@testing-library/react/cleanup-after-each';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import MyChildComponent from '../components/MyChildComponent';

afterEach(cleanup);

describe('<MyChildComponent />', () => {
  it('renders as expected', async () => {
    const { container } = render(<MyChildComponent />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders with props as expected', async () => {
    const { container } = render(<MyChildComponent msg="Test" open={false} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
