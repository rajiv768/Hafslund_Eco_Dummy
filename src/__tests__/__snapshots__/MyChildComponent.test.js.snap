// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MyChildComponent /> renders as expected 1`] = `
<div
  class="ui segment"
>
  <h3>
    Hello World
  </h3>
  <h5>
    Widget Open
  </h5>
</div>
`;

exports[`<MyChildComponent /> renders with props as expected 1`] = `
<div
  class="ui segment"
>
  <h3>
    Test
  </h3>
</div>
`;
